web/src/message_notifications.ts

```diff
@@ -375,12 +410,12 @@ export function should_send_audible_notification(message) {
     return false;
 }
 
-export function received_messages(messages) {
+export function received_messages(messages: (Message | TestNotificationMessage)[]): void {
     for (const message of messages) {
         if (!message_is_notifiable(message)) {
             continue;
         }
-        if (!unread.message_unread(message)) {
+        if (!message.unread) {
```

Comment:
This is a dangerous type of thing to change hidden in a TS migration commit. It's worth noting that it refetches the message object from `message_store`, so this code change's correctness is relying on having a fresh copy of the object, which is valid because we use a singleton pattern for these, but that sorta is a complex global reasoning thing, not a mechanical translation like the rest of this commit.

I think `unread.message_unread` dated from when we were not maintaining the `message.unread` field on Message objects; since we are, it seems possible we should clean up the other calls to that and delete the abstraction. But this should be its own commit.

Reply:
Thanks for pointing that out! I really thought I'd left a comment here asking if this was okay or if I should do something similar to `small_avatar_url_for_test_notification` above, but I guess not.

Do you feel confident that we're maintaining the `unread` field now and can delete the abstraction? I don't know enough about the uses of `unread` to feel confident myself. I'll change this for now to still use `message_unread` function for the `Message` object, but I'm happy to do a followup to delete that abstraction.

Reply:
Wait, I'm confused.

```js
export function message_unread(message: Message): boolean {
    if (message === undefined) {
        return false;
    }
    return message.unread;
}
```

it doesn't look like this refetches the message? So I think it should be fine?

Reply:
Yeah it looks like that helper is useless.

Reply:
Added a commit to remove that helper